plugin/health/health.go

```diff
@@ -61,6 +61,17 @@ func (h *health) OnStartup() error {
 	return nil
 }
 
+func (h *health) OnRestart() error {
+	// relingish our listener as we re-listen on successfull reload
+	if h.ln != nil {
+		if err := h.ln.Close(); err != nil {
+			return err
+		}
+		h.ln = nil
+	}
+	return nil
+}
+
 func (h *health) OnShutdown() error {
 	// Stop polling plugins
 	h.pollstop <- true
```

Comment:
The lameduck part should happen on RESTART ?
I would think only on FINAL shutdown.

That means maybe having a separated function to close Listener and to process the Lameduck.

OnRestart -> close listener
OnFinalShutdown -> close listener + lameduck delay